### ISSUE TITLE
oversubscribing on openmpi

### ISSUE THREAD

**Issue by Caylo** [hpcugent/vsc-mympirun#96]
openmpi doesnt allow oversubscribing, in some cases this causes issues when using --double.
see https://www.open-mpi.org/faq/?category=running#oversubscribing

**Comment by boegel** [comment]
follow-up on changes in #93

**Comment by boegel** [comment]
@Caylo Can you be a bit more specific w.r.t. which issues and when exactly this would occur?

**Comment by Caylo** [comment]
it happens when you specify a number of processes that is larger than what is available (which is pretty much always when using `--double` as far as i can tell). the issue is that openmpi doesn't like oversubscribing and gives an error message 

```
(nodes=2:ppn=4)
-$ mympirun --double hostname
--------------------------------------------------------------------------
There are not enough slots available in the system to satisfy the 16 slots 
that were requested by the application:
  hostname

Either request fewer slots for your application, or make more slots available
for use.
--------------------------------------------------------------------------
```
the documentation specifies a way around it using `slots` in the hostfile, but we still have to implement this into mympirun somehow

**Comment by boegel** [comment]
OK, that's helpful. We should fix this at some point, but it's low priority since i) `--double` isn't used very often, and ii) most of our HPC-UGent users use Intel MPI rather than OpenMPI.

**Comment by boegel** [comment]
fixed in #125, #127
